### PR TITLE
Fe/fix swap issues

### LINKED ISSUE #168: Swaps query might be causing positions issues
**Describe the bug**
- Folks who have manually minted + sold have reporting seeing incorrect short values 
- Someone who went long isn't able to see their position on the dashboard 

**To Reproduce**
1. Go short, manually rebalance (mint + sell more), close position 
2. Go long 

**Expected behavior**
Should show the accurate positions and positions should not disappear 

**Screenshots**
Screen Shot 2022-02-15 at 12 26 26 AM
Screen Shot 2022-02-15 at 12 27 58 AM
Screen Shot 2022-02-15 at 12 28 08 AM

### PR DESCRIPTION
**Pull request by yichiehliu**
# Task:

## Description

1. add useVaultHistory to get the correct minted balance and openShort balance in the first valid vault
2. need to merge subgraph PR #125  first to get it on mainnet, now I use https://thegraph.com/hosted-service/subgraph/yichiehliu/squeeth for ropsten

Fixes #168  (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Document update

## How Has This Been Tested

Locally

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] Added video recordings if it is a UI change

### CODE REVIEW

File: packages/frontend/src/context/positions.tsx [hunk 1]
```diff
@@ -108,9 +109,10 @@ const PositionsProvider: React.FC = ({ children }) => {
       : new BigNumber(0)
   }, [firstValidVault, oSqueethBal.toString(), positionType, shortVaults?.length, squeethAmount.toString()])
 
-  const shortDebt = useMemo(() => {
-    return positionType === PositionType.SHORT ? squeethAmount : new BigNumber(0)
-  }, [positionType, squeethAmount.toString()])
+  //meaning if there is minted squeeth sold, need to be subtracted
+  const finalShortSqueeth = useMemo(() => {
+    return mintedSqueeth > mintedDebt ? openShortSqueeth.plus(mintedSqueeth).minus(mintedDebt) : openShortSqueeth
```
Comment (alexisgauba) [inline packages/frontend/src/context/positions.tsx:114]: what is the difference between mintedSqueeth and mintedDebt? Is mintedSqueeth the total amount of all squeeth minted and mintedDebt the amount of minted squeeth that's been sold?

File: packages/frontend/src/hooks/useVaultHistory.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      },
+      updateQuery(prev, { subscriptionData }) {
+        if (!subscriptionData.data) return prev
+        const newVaultsHistories = subscriptionData.data.vaultHistories
+        return { vaultHistories: newVaultsHistories }
+      },
+    })
+  }, [address, vaultId, subscribeToMore])
+
+  //accumulated four actions, mintedSqueeth doesn't take minted squeeth sold into account
```
Comment (alexisgauba) [inline packages/frontend/src/hooks/useVaultHistory.ts:41]: ok so mintedsqueeth is minted and not sold or LPed? and openShortSqueeth is any squeeth that was minted + short regardless if manual or through shortHelper?

File: packages/frontend/src/utils/apollo-client.ts [hunk 3]
```diff
@@ -11,7 +11,7 @@ const httpLinkRP = new HttpLink({
 })
 
 const httpLinkRPSqueeth = new HttpLink({
-  uri: 'https://api.thegraph.com/subgraphs/name/opynfinance/squeeth-ropsten',
+  uri: 'https://api.thegraph.com/subgraphs/name/yichiehliu/squeeth',
```
Comment (alexisgauba) [inline packages/frontend/src/utils/apollo-client.ts:14]: Can we put this deployment under opyn? Or were there issues w doing that?

**Review by alexisgauba — COMMENTED** [review]
I'm not

File: packages/frontend/src/utils/apollo-client.ts (hunk 3 above)
Comment (yichiehliu) [reply packages/frontend/src/utils/apollo-client.ts:14]: Yh can do it after #125 merged, now only mine has the updated action types

**Review by alexisgauba — CHANGES_REQUESTED** [review]
Hmm it's not showing my short position. Would recommend listing out and testing important scenarios. Eg. some scenarios: 
- go short through short helper 
- go short manually
- go short through short helper and then manually as well 
- go short manually and then through short helper 
- go short and then have extra minted 
- go short and LP 

This PR: 
Screen Shot 2022-02-20 at 6 53 11 PM

squeeth.com:
Screen Shot 2022-02-20 at 6 55 57 PM

File: packages/frontend/src/hooks/useVaultHistory.ts (hunk 2 above)
Comment (yichiehliu) [reply packages/frontend/src/hooks/useVaultHistory.ts:41]: OpenShortSqueeth & mintedSqueeth are directly calculated from vault history, and so mintedSqueeth doesn't take sold or LPed into account

And mintedDebt is calculated from contract & swaps data
So I have finalOpenShort, compated mintedSqueeth & mintedDebt to get correct short amount including manually selling, if mintedSqueeth > mintedDebt means that users manually sell their minted oSQTH, so the openShort + mintedSqueeth - mintedDebt = actual short position

**Pull request closed without merging**